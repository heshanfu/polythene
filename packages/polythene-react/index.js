export * from "polythene-react-base";
export * from "polythene-react-button";
export * from "polythene-react-card";
export * from "polythene-react-checkbox";
export * from "polythene-react-dialog";
export * from "polythene-react-dialog-pane";
export * from "polythene-react-drawer";
export * from "polythene-react-fab";
export * from "polythene-react-icon";
export * from "polythene-react-icon-button";
export * from "polythene-react-icon-label-button";
export * from "polythene-react-ios-spinner";
export * from "polythene-react-list";
export * from "polythene-react-list-tile";
export * from "polythene-react-material-design-progress-spinner";
export * from "polythene-react-material-design-spinner";
export * from "polythene-react-menu";
export * from "polythene-react-notification";
export * from "polythene-react-radio-button";
export * from "polythene-react-radio-group";
export * from "polythene-react-raised-button";
export * from "polythene-react-ripple";
export * from "polythene-react-search";
export * from "polythene-react-shadow";
export * from "polythene-react-slider";
export * from "polythene-react-snackbar";
export * from "polythene-react-svg";
export * from "polythene-react-switch";
export * from "polythene-react-tabs";
export * from "polythene-react-textfield";
export * from "polythene-react-toolbar";
