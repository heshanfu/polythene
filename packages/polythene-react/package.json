{
  "name": "polythene-react",
  "version": "1.1.0",
  "description": "",
  "main": "dist/polythene-react",
  "module": "dist/polythene-react.mjs",
  "scripts": {
    "clean": "rm -rf dist/*",
    "rollup": "../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd.js && ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.es.js",
    "build-standalone": "DEPS=1 ENTRY=scripts/standalone.js SOURCEMAP=1 DEST=./dist/polythene-react-standalone ../../node_modules/rollup/bin/rollup -c ../../scripts/rollup.umd-includes.js",
    "build": "yarn run clean && yarn run rollup && yarn run build-standalone"
  },
  "files": [
    "dist",
    "README.md"
  ],
  "dependencies": {
    "polythene-css": "^1.1.0",
    "polythene-react-base": "^1.1.0",
    "polythene-react-button": "^1.1.0",
    "polythene-react-card": "^1.1.0",
    "polythene-react-checkbox": "^1.1.0",
    "polythene-react-dialog": "^1.1.0",
    "polythene-react-dialog-pane": "^1.1.0",
    "polythene-react-drawer": "^1.1.0",
    "polythene-react-fab": "^1.1.0",
    "polythene-react-icon": "^1.1.0",
    "polythene-react-icon-button": "^1.1.0",
    "polythene-react-icon-label-button": "^1.0.0",
    "polythene-react-ios-spinner": "^1.1.0",
    "polythene-react-list": "^1.1.0",
    "polythene-react-list-tile": "^1.1.0",
    "polythene-react-material-design-progress-spinner": "^1.1.0",
    "polythene-react-material-design-spinner": "^1.1.0",
    "polythene-react-menu": "^1.1.0",
    "polythene-react-notification": "^1.1.0",
    "polythene-react-radio-button": "^1.1.0",
    "polythene-react-radio-group": "^1.1.0",
    "polythene-react-raised-button": "^1.1.0",
    "polythene-react-ripple": "^1.1.0",
    "polythene-react-search": "^1.1.0",
    "polythene-react-shadow": "^1.1.0",
    "polythene-react-slider": "^1.1.0",
    "polythene-react-snackbar": "^1.1.0",
    "polythene-react-svg": "^1.1.0",
    "polythene-react-switch": "^1.1.0",
    "polythene-react-tabs": "^1.1.0",
    "polythene-react-textfield": "^1.1.0",
    "polythene-react-toolbar": "^1.1.0"
  },
  "devDependencies": {
    "polythene-core": "^1.1.0",
    "polythene-core-css": "^1.1.0",
    "react": "^16.2.0",
    "react-hyperscript": "^3.0.0"
  },
  "author": "Arthur Clemens <arthurclemens@gmail.com> (http://arthurclemens.com)",
  "license": "MIT"
}
