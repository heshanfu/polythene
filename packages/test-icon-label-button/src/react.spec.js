import { runSnapshots } from "../../../scripts/tests/react-snapshots";
import { renderer, keys, IconLabelButton } from "polythene-react";
import specTests from "./spec-tests.js";
import reactTests from "./tests-react.js";

runSnapshots({
  tests: specTests({ IconLabelButton, renderer, keys }).concat(reactTests),
  renderer
});
