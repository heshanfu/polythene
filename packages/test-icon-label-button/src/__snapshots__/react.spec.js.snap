exports[`test Dark tone class + light tone 1`] = `
<div
  style={
    Object {
      "background": "#fff",
      "padding": "20px",
    }
  }>
  <Checkbox
    label="Label"
    tone="light" />
</div>
`;

exports[`test Dark tone class + light tone class 1`] = `
<div
  className="pe-light-tone"
  style={
    Object {
      "background": "#fff",
      "padding": "20px",
    }
  }>
  <Checkbox
    label="Label" />
</div>
`;

exports[`test No options 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: class 1`] = `
<SelectionControl
  className="class-x"
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: className 1`] = `
<SelectionControl
  className="className-x"
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: defaultChecked (JSX) 1`] = `
<Checkbox
  defaultChecked={true}
  label="Label" />
`;

exports[`test Option: defaultChecked -- dark tone class 1`] = `
<SelectionControl
  defaultChecked={true}
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: defaultChecked 1`] = `
<SelectionControl
  defaultChecked={true}
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: disabled -- dark tone class 1`] = `
<div
  className="multiple">
  <Checkbox
    disabled={true}
    label="Off" />
  <Checkbox
    defaultChecked={true}
    disabled={true}
    label="On" />
</div>
`;

exports[`test Option: disabled 1`] = `
<div
  className="multiple">
  <Checkbox
    disabled={true}
    label="Off" />
  <Checkbox
    defaultChecked={true}
    disabled={true}
    label="On" />
</div>
`;

exports[`test Option: element 1`] = `
<SelectionControl
  element="a"
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: iconButton (custom hover behaviour) 1`] = `
<SelectionControl
  iconButton={
    Object {
      "ink": false,
      "wash": true,
    }
  }
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: iconOn, iconOff (custom icon) (JSX) 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }>
  <Checkbox
    iconOff={
      Object {
        "svg": Object {
          "content": <svg
            enableBackground="new 0 0 24.00 24.00"
            height="24"
            viewBox="0 0 24.00 24.00"
            width="24">
            <path
                  d="M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z "
                  fill="#000000"
                  fillOpacity="1"
                  strokeLinejoin="round"
                  strokeWidth="0.2" />
      </svg>,
        },
      }
    }
    iconOn={
      Object {
        "svg": Object {
          "content": <svg
            height="24"
            viewBox="0 0 24 24"
            width="24">
            <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z" />
      </svg>,
        },
      }
    }
    label="Label" />
</div>
`;

exports[`test Option: iconOn, iconOff (custom icon) 1`] = `
<div
  className="multiple">
  <Checkbox
    iconOff={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                      }
            } />,
        },
      }
    }
    iconOn={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                      }
            } />,
        },
      }
    }
    label="small"
    size="small" />
  <Checkbox
    iconOff={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                      }
            } />,
        },
      }
    }
    iconOn={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                      }
            } />,
        },
      }
    }
    label="regular"
    size="regular" />
  <Checkbox
    iconOff={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                      }
            } />,
        },
      }
    }
    iconOn={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                      }
            } />,
        },
      }
    }
    label="medium"
    size="medium" />
  <Checkbox
    iconOff={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24.00 24.00\" enable-background=\"new 0 0 24.00 24.00\"><path fill=\"#000000\" fill-opacity=\"1\" stroke-width=\"0.2\" stroke-linejoin=\"round\" d=\"M 11.9994,15.3943L 8.2364,17.6643L 9.2314,13.3833L 5.9094,10.5053L 10.2894,10.1293L 11.9994,6.09327L 13.7094,10.1293L 18.0894,10.5053L 14.7674,13.3833L 15.7624,17.6643M 21.9994,9.24227L 14.8084,8.62526L 11.9994,1.99827L 9.1904,8.62526L 1.9994,9.24227L 7.4544,13.9693L 5.8194,20.9983L 11.9994,17.2703L 18.1794,20.9983L 16.5444,13.9693L 21.9994,9.24227 Z \"/></svg>",
                      }
            } />,
        },
      }
    }
    iconOn={
      Object {
        "svg": Object {
          "content": <div
            dangerouslySetInnerHTML={
                  Object {
                        "__html": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z\"/></svg>",
                      }
            } />,
        },
      }
    }
    label="large"
    size="large" />
</div>
`;

exports[`test Option: id 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  id="id-x"
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: label 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  label="Label"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Option: selectable 1`] = `
<div
  className="multiple">
  <Checkbox
    label="Never"
    selectable={[Function]} />
  <Checkbox
    label="Only when unchecked"
    selectable={[Function]} />
</div>
`;

exports[`test Option: size 1`] = `
<div
  className="multiple">
  <Checkbox
    label="small"
    size="small" />
  <Checkbox
    label="regular"
    size="regular" />
  <Checkbox
    label="medium"
    size="medium" />
  <Checkbox
    label="large"
    size="large" />
</div>
`;

exports[`test Option: tabindex 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  selectable={[Function]}
  tabIndex={3}
  type="checkbox" />
`;

exports[`test Option: value 1`] = `
<SelectionControl
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  name="worth"
  selectable={[Function]}
  type="checkbox"
  value="notable" />
`;

exports[`test Styled Checkbox 1`] = `
<SelectionControl
  defaultChecked={true}
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  label="Label"
  selectable={[Function]}
  style={
    Object {
      "color": "#ff0000",
    }
  }
  type="checkbox" />
`;

exports[`test Themed Checkbox (color and font size) 1`] = `
<SelectionControl
  className="tests-checkbox-themed-checkbox"
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  label="Label"
  selectable={[Function]}
  type="checkbox" />
`;

exports[`test Themed Checkbox (colors) -- dark tone class 1`] = `
<SelectionControl
  className="tests-checkbox-themed-checkbox"
  icons={
    Object {
      "iconOff": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\"/></svg>",
      "iconOn": "<svg width=\"24\" height=\"24\" viewBox=\"0 0 24 24\"><path d=\"M19 3H5c-1.11 0-2 .9-2 2v14c0 1.1.89 2 2 2h14c1.11 0 2-.9 2-2V5c0-1.1-.89-2-2-2zm-9 14l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z\"/></svg>",
    }
  }
  instanceClass="pe-checkbox-control"
  label="Label"
  selectable={[Function]}
  type="checkbox" />
`;
